x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/Converter.java

```diff
@@ -9,7 +9,18 @@
 
 import org.elasticsearch.common.io.stream.NamedWriteable;
 
+import java.time.ZoneId;
+
 public interface Converter extends NamedWriteable {
 
-    Object convert(Object value);
+    default Object convert(Object value) {
+        return convert(value, null);
+    }
+
+    /**
+     * Converts `value` to the target type considering `zoneId` if needed.
+     * `zoneId` can be null and it is ignored for all non-timezone aware conversions.
+     */
+    Object convert(Object value, ZoneId zoneId);
```

Comment:
I've also investigated whether the `zoneId` could be stored in the converter but this would be a lot more invasive change.

Reply:
Indeed, adding a specific method for one specific data type doesn't feel ideal, but still better than instantiating the converter per session.